x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/PassThroughOperator.java

```diff
@@ -44,5 +44,10 @@ public boolean isFinished() {
     }
 
     @Override
-    public void close() {}
+    public void close() {
+        if (page != null) {
+            // this can only happen in a failure situation, prior to the operator being exhausted
+            page.releaseBlocks();
+        }
+    }
```

Comment:
This might should just extend `AbstractPageMappingOperator` to get that for free.

Reply:
I'll take a look in a follow-up. I need to revisit the test that uses these as it is awaits fix :-(